azuredevops/internal/service/core/resource_team.go

```diff
@@ -0,0 +1,558 @@
+	if team == nil {
+		d.SetId("")
+		return
+	}
+
+	d.SetId(team.Id.String())
+	d.Set("name", team.Name)
+	d.Set("description", team.Description)
+	d.Set("administrators", administrators)
+	d.Set("members", members)
+}
+
+func readTeamMembers(clients *client.AggregatedClient, team *core.WebApiTeam) (*schema.Set, error) {
+	members, err := clients.IdentityClient.ReadMembers(clients.Ctx, identity.ReadMembersArgs{
+		ContainerId: converter.String(team.Id.String()),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return readSubjectDescriptors(clients, members)
+}
+
+func updateTeamMembers(clients *client.AggregatedClient, team *core.WebApiTeam, subjectDescriptors *[]string) error {
+	var err error
+
+	currentMemberSet, err := readTeamMembers(clients, team)
+	if err != nil {
+		return err
+	}
+	if (subjectDescriptors == nil || len(*subjectDescriptors) <= 0) && currentMemberSet.Len() <= 0 {
+		return nil
+	}
+
+	currentMembers := currentMemberSet.List()
+
+	// determine the list of all removed members
+	err = removeTeamMembers(clients, team, linq.From(currentMembers).Except(linq.From(*subjectDescriptors)))
+	if err != nil {
+		return err
+	}
+
+	// determine the list of all added members
+	err = addTeamMembers(clients, team, linq.From(*subjectDescriptors).Except(linq.From(currentMembers)))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getIdentitiesFromSubjects(clients *client.AggregatedClient, query linq.Query) (*[]identity.Identity, error) {
+	if !query.Any() {
+		return &[]identity.Identity{}, nil
+	}
+
+	discriptors := query.
+		Aggregate(func(r interface{}, i interface{}) interface{} {
+			if r.(string) == "" {
+				return i
+			}
+			return r.(string) + "," + i.(string)
+		}).(string)
+
+	idlist, err := clients.IdentityClient.ReadIdentities(clients.Ctx, identity.ReadIdentitiesArgs{
+		SubjectDescriptors: converter.String(discriptors),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return idlist, err
+}
+
+func removeTeamMembers(clients *client.AggregatedClient, team *core.WebApiTeam, query linq.Query) error {
+	idList, err := getIdentitiesFromSubjects(clients, query)
+	if err != nil {
+		return err
+	}
+
+	for _, id := range *idList {
+		log.Printf("[TRACE] Removing member %s from team %s", id.Id.String(), *team.Name)
+
+		_, err := clients.IdentityClient.RemoveMember(clients.Ctx, identity.RemoveMemberArgs{
+			ContainerId: converter.String(team.Id.String()),
+			MemberId:    converter.String(id.Id.String()),
+		})
+		if err != nil {
+			return fmt.Errorf("Error removing member %s from team %s: %+v", id.Id.String(), *team.Name, err)
+		}
+	}
+	return nil
+}
+
+func addTeamMembers(clients *client.AggregatedClient, team *core.WebApiTeam, query linq.Query) error {
+	idList, err := getIdentitiesFromSubjects(clients, query)
+	if err != nil {
+		return err
+	}
+	if idList == nil || len(*idList) != query.Count() {
+		return fmt.Errorf("Failed to load identity data for subjects")
```

Comment:
For such a detailed error message I must determine the difference between what's expected and what has been returned by the API. I'm not sure if it's really worth doing so, especially because when such a thing happens something is completely off inside Azure DevOps and thus you might need support from Microsoft to get it analyzed and fixed. Well at least I guess so.